.github/workflows/main.yml

```diff
@@ -6,12 +6,12 @@ jobs:
   Test:
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Go 1.17.x
-        uses: actions/setup-go@v4
-        with:
-          go-version: '1.17.x'
       - name: Checkout
         uses: actions/checkout@v3
+      - name: Set up Go
```

Comment:
☝️ Note that it's necessary to `checkout` _before_ `setup-go`-ing, such that the `go.mod` file is present on the actions runner, if that's not clear.